lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_workspace_runtime.rb

```diff
@@ -52,13 +52,56 @@ def self.instantiate_with_user(uri, user, attrs)
       workspace = MiqAeWorkspaceRuntime.new(attrs)
       self.current = workspace
       workspace.instantiate(uri, user, nil)
+
+      if !uri.nil?
+        _scheme, _userinfo, _host, _port, _registry, _path, _opaque, query, _fragment = MiqAeUri.split(uri, "miqaedb")
+        args = MiqAeUri.query2hash(query)
+        miq_request_id = find_miq_request_id(args)
+      else
+        miq_request_id = nil
+      end
+
       workspace
     rescue MiqAeException => err
-      $miq_ae_logger.error(err.message)
+      $miq_ae_logger.error(err.message, :resource_id => miq_request_id)
     ensure
       clear_stored_workspace
     end
 
+    def self.find_miq_request_id(args)
+      miq_request_id = nil
+      if args['MiqRequest::miq_request']
+        miq_request_id = args['MiqRequest::miq_request']
+      elsif args['vmdb_object_type'].to_s.include?("task")
```

Comment:
Can you verify the vmdb_object_type is lowercase all the time?

Reply:
yeah just checked and can confirm that `vmdb_object_type` is always lowercase. It always looks something like `vm_retire_task`, `vm_retire_request`, or `vm`